### PR TITLE
Backend PyTorch supports forward-mode automatic differentiation: Step 1

### PR DESCRIPTION
**Pull request by ZongrenZou**

### CODE REVIEW

File: deepxde/model.py [hunk 1]
```diff
@@ -301,7 +301,12 @@ def outputs_losses(training, inputs, targets, auxiliary_vars, losses_fn):
             # Data losses
             if targets is not None:
                 targets = torch.as_tensor(targets)
-            losses = losses_fn(targets, outputs_, loss_fn, inputs, self)
+            if config.autodiff == "reverse":
+                aux = None
+            elif config.autodiff == "forward":
+                # if forward-mode AD is used, then a forward call needs to be passed
+                aux = [lambda _x: self.net(_x)]
```
Comment (lululxvi) [inline deepxde/model.py:308]:
```
aux = None
if config.autodiff == "forward":
    # if forward-mode AD is used, then a forward call needs to be passed
    aux = [lambda _x: self.net(_x)]
```

Comment (ZongrenZou) [reply deepxde/model.py]: Done.

File: deepxde/model.py [hunk 2]
```diff
@@ -301,7 +301,9 @@ def outputs_losses(training, inputs, targets, auxiliary_vars, losses_fn):
             # Data losses
             if targets is not None:
                 targets = torch.as_tensor(targets)
-            losses = losses_fn(targets, outputs_, loss_fn, inputs, self)
+            # if forward-mode AD is used, then a forward call needs to be passed
+            aux = [lambda _x: self.net(_x)] if config.autodiff == "forward" else None
```
Comment (lululxvi) [inline deepxde/model.py:305]:
```
aux = [self.net] if config.autodiff == "forward" else None
```

Comment (ZongrenZou) [reply deepxde/model.py]: Done.

**Pull request merged**